src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+/**
+ * Parser class deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Parses user input and carry out commands accordingly.
+     *
+     * @param tasklist list of tasks in Duke application.
+     * @param ui user interface of Duke application.
+     * @param storage stores the tasklist into the hard drive.
+     */
+    public static void parse(TaskList tasklist, Ui ui, Storage storage) {
+        while (ui.hasNextLine()) {
+            String input = ui.getUserCommand();
+            try {
+                //  Bye
+                if (input.equals("bye")) {
+                    ui.sayBye();
+                    return;
+                }
+                //  List
+                if (input.equals("list")) {
+                    ui.list(tasklist);
+                    continue;
+                }
+                //  Marking
+                if ((input.length() >= 4) && (input.substring(0, 4).equals("mark"))) {
+                    //  error checking
+                    if (input.length() == 4 || input.substring(5).equals("")) {
+                        throw new IllegalIndexException("OOPS!!! The index of a mark cannot be empty.");
+                    }
+                    String remainder = input.substring(5);
+                    int index = Integer.valueOf(remainder) - 1;
+                    tasklist.mark(index, ui);
+                    storage.updateStorage(tasklist);
+                    continue;
+                }
```

Comment:
I like that you used `continue` instead of if-else blocks. I think it makes the code easier to read!